test/minikube/minikube_base_database_test.go

```diff
@@ -142,51 +142,51 @@ func verifyEngineAltAddress(t *testing.T, namespaceName string, admin0 string, e
 }
 
 func backupDatabase(t *testing.T, namespaceName string, podName string, databaseName string, options *helm.Options) {
-	randomSuffix := strings.ToLower(random.UniqueId())
-
-	bakName := fmt.Sprintf("backup-full-%s", randomSuffix)
-
-	kubectlOptions := k8s.NewKubectlOptions("", "")
-	options.KubectlOptions = kubectlOptions
-	options.KubectlOptions.Namespace = namespaceName
-
-	testlib.AddTeardown(testlib.TEARDOWN_BACKUP, func() { helm.Delete(t, options, bakName, true) })
-	helm.Install(t, options, testlib.BACKUP_HELM_CHART_PATH, bakName)
 
 	// wait for the backup to both start _and_ complete successfully
-	backupJob := fmt.Sprintf("backup-%s-job-full", databaseName)
+	backupJob := fmt.Sprintf("hotcopy-%s-job-initial", databaseName)
 	testlib.AwaitPodPhase(t, namespaceName, backupJob, corev1.PodSucceeded, 120*time.Second)
 
 	// verify that the backup has been documented by the Admin layer
-	backupName := fmt.Sprintf("%s/%s", testlib.LAST_BACKUP_PREFIX, databaseName)
+	// backupName := fmt.Sprintf("%s/%s", testlib.LAST_BACKUP_PREFIX, databaseName)
 	backupset, err := k8s.RunKubectlAndGetOutputE(t, options.KubectlOptions,
 		"exec", podName, "--",
-		"nuocmd", "get", "value",
-		"--key", backupName,
+		"nuodocker", "get", "current-backup", "--db-name", databaseName,
 	)
 
-	assert.NilError(t, err, "Error running: nuocmd get value --key ", backupName)
+	assert.NilError(t, err, "Error running: nuodocker get current-backup  ")
 	assert.Check(t, backupset != "")
 }
 
-func restoreDatabase(t *testing.T, namespaceName string) {
+func restoreDatabase(t *testing.T, namespaceName string, podName string, databaseOptions *helm.Options, restart string) {
```

Comment:
`restart` is always true, so all of that `restore` code is unused. Also GoLang has a `bool` type, no need to compare a string to `"true"`

Reply:
I was making the function reasonably general purpose.
So restart is only "true" if the caller sets it so.

I passed in the string "true" rather than the bool to make the assignment to "restore.autoRestart" simpler and cleaner - because Go has no trinary operator.

Reply:
I am not a fan of function that are general purpose today, but might break in the future. You are also only ever checking true and nothing else.

`restore()` is not called when it is not true.

Reply:
Having said that about converting from bool to string being less simple - I can see a way to do it now.

So I might change the method signature - the purist in me, and in us all... :)

Reply:
And having said _all_ of the above - I am not a big fan of bool parameters to methods, because they are so easily misunderstood.

Reply:
That is what `enums` are for ;)

Reply:
The code:
```
else {
   restore()
}
```
calls restore() when the value is not true...

Reply:
But I am happy to remove that complication until it is needed.

Reply:
> That is what enums are for ;)
I went on a jagg to replace all uses of string args and Hashtable keys in Java with Enums - wherever possible - so as to enlist the compiler in detecting typos, rather than the runtime.

The result was generally good - but various, frequently-used, pieces of code became clumsy as a result...